domain.go

```diff
@@ -1720,6 +1720,15 @@ func (d *Domain) MemoryStats(nrStats uint32, flags uint32) ([]DomainMemoryStat,
 	return out, nil
 }
 
+// See also https://libvirt.org/html/libvirt-libvirt-domain.html#virDomainGetConnect
+func (d *Domain) DomainGetConnect() (*Connect, error) {
+	ptr := C.virDomainGetConnect(d.ptr)
+	if ptr == nil {
+		return nil, GetLastError()
+	}
+	return &Connect{ptr: ptr}, nil
```

Comment:
The C API has rather dangerous semantics in that it does *not* acquire a reference on the returned connection.

I think it is desirable to fix that in the Go binding by calling virConnectRef.